src/Form/YamlType.php

```diff
@@ -38,7 +38,11 @@ function ($optionsAsString) {
                 }
 
                 try {
-                    return Yaml::parse($optionsAsString);
+                    $val = Yaml::parse($optionsAsString);
+                    if (!is_array($val)) {
+                        throw new ParseException('Result is not a array');
+                    }
+                    return $val;
```

Comment:
Blank line before return, to comply with Sf standards
```suggestion

                    return $val;
```